workspaces/src/network/mainnet.rs

```diff
@@ -0,0 +1,90 @@
+use std::path::PathBuf;
+use std::str::FromStr;
+
+use async_trait::async_trait;
+use url::Url;
+
+use near_primitives::views::FinalExecutionStatus;
+
+use crate::network::Info;
+use crate::network::{
+    Account, CallExecution, CallExecutionDetails, NetworkClient, NetworkInfo,
+    TopLevelAccountCreator,
+};
+use crate::rpc::{client::Client, tool};
+use crate::types::{AccountId, InMemorySigner, SecretKey};
+use crate::Contract;
+
+const RPC_URL: &str = "https://rpc.mainnet.near.org";
+const HELPER_URL: &str = "https://helper.mainnet.near.org";
+
+pub struct Mainnet {
+    client: Client,
+    info: Info,
+}
+
+impl Mainnet {
+    pub(crate) fn new() -> Self {
+        Self {
+            client: Client::new(RPC_URL.into()),
+            info: Info {
+                name: "mainnet".into(),
+                root_id: AccountId::from_str("near").unwrap(),
+                keystore_path: PathBuf::from(".near-credentials/mainnet/"),
+                rpc_url: RPC_URL.into(),
+            },
+        }
+    }
+}
+
+#[async_trait]
+impl TopLevelAccountCreator for Mainnet {
+    async fn create_tla(
+        &self,
+        id: AccountId,
+        sk: SecretKey,
```

Comment:
Do we really need to pass SK here? Since we have this path `.near-credentials/mainnet/`, can we read it from the filesystem?

Reply:
this function was just to be generic over how we could create top level accounts, because a user can grab credentials from just about anywhere. So if they wanted to read from `.near-credentials/mainnet/`, they have to do it manually themselves to be explicit or we offer a convenience function down the line